### PR TITLE
Fix precision loss deprecation on float to int conversion

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Related to #4883

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by MGatner — APPROVED** [review]
So basically we are embracing the loss and circumventing the warning?

File: system/Images/Handlers/GDHandler.php [hunk 1]
```diff
@@ -472,9 +472,9 @@ protected function textOverlay(string $text, array $options = [], bool $isShadow
         // Add the shadow to the source image
         if (! empty($options['fontPath'])) {
             // We have to add fontheight because imagettftext locates the bottom left corner, not top-left corner.
-            imagettftext($src, $options['fontSize'], 0, $xAxis, $yAxis + $options['fontheight'], $color, $options['fontPath'], $text);
+            imagettftext($src, $options['fontSize'], 0, (int) $xAxis, $yAxis + $options['fontheight'], $color, $options['fontPath'], $text);
```
Comment (sfadschm) [inline system/Images/Handlers/GDHandler.php:475]: Is the yAxis automatically cast due to the '+'?

**Review by kenjis — APPROVED** [review]

File: system/Images/Handlers/GDHandler.php (hunk 1 above)
Comment (paulbalandan) [reply system/Images/Handlers/GDHandler.php]: Good catch! Fixed

**Comment by paulbalandan** [comment]
> So basically we are embracing the loss and circumventing the warning?

Basically, yes. And we're just making it explicit now. Previously, because of loose typing, float got coerced to int implicitly but because this behavior is now deprecated, we're making it explicit to match with existing bahavior. If ever this would be buggy in the future, then we can adjust later.

**Pull request merged**